pyvisa_py/sessions.py

```diff
@@ -122,8 +122,8 @@ def close(self) -> StatusCode:
     #: Session type as (Interface Type, Resource Class)
     session_type: Tuple[constants.InterfaceType, str]
 
-    #: Timeout in seconds to use when opening the resource.
-    open_timeout: Optional[float]
+    #: Timeout in milliseconds to use when opening the resource.
+    open_timeout: Optional[int]
```

Comment:
I will need to check how that interact with other backends (which are far harder to test too).

Reply:
From my examination of the code, this appears to have been a bug.  The open_timeout is passed in from VISA where it is always expressed as an int in milliseconds (see constants.VI_TMO_IMMEDIATE and constants.VI_TMO_INFINITE).

Reply:
I think you are indeed right thanks for the fix.